feat: Exponential backoff factor for HTTP request retries (#3358)
fubuloubu/web3.py: docs/internals.rst

``HTTPProvider`` and ``AsyncHTTPProvider`` instances retry certain requests by default
on exceptions. This can be configured via configuration options on the provider
instance. The retry mechanism employs an exponential backoff strategy, starting from
the initial value determined by the ``backoff_factor``, and doubling the delay with

> each attempt, up to the ``retries`` value.Below is an example showing the default

Reviewer: Just one tiny nit to wrap this up. Thanks for the quick replies 👌🏼 

```suggestion
each attempt, up to the ``retries`` value. Below is an example showing the default
```

``HTTPProvider`` and ``AsyncHTTPProvider`` instances retry certain requests by default
on exceptions. This can be configured via configuration options on the provider

> instance. Retry backoff time would be ``backoff_factor * 2**retry_times``. Below is an

Reviewer: I think we can be a bit less analytical here, use a touch more words, and explain that it doubles each time. This should help give a bit clearer picture on what sort of intervals the user would want to configure to and perhaps why we went with a number like `0.125` as the default.

```suggestion
instance. The retry mechanism employs an exponential backoff strategy, starting from the initial value determined by the ``backoff_factor``, and doubling the delay with each attempt, up to the ``retries`` value. Below is an
```

            # number of retries to attempt
            retries=5,

>             # time interval between retries would be backoff_factor * 2**retry_times

Reviewer: I don't think the technical aspect is super important here either. A curious user can look at the code to see the implementation.

```suggestion
            # initial delay multiplier, doubles with each retry attempt
```

>             backoff_factor=0.5,

Reviewer: This should also be `0.125`

```suggestion
            backoff_factor=0.125,
```
